tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -46,6 +46,23 @@ spec:
             exit 1
         fi
 
+        # images are required to use sha reference - check this
+        CHECK_RESULT=success
+        NUM_COMPONENTS=$(jq '.components | length' "$SNAPSHOT_SPEC_FILE")
+        for ((i=0; i < NUM_COMPONENTS; i++)); do
+          COMPONENT=$(jq -c ".components[$i]" "$SNAPSHOT_SPEC_FILE")
+          NAME=$(jq -r '.name' <<< "$COMPONENT")
+          IMAGE_REF=$(jq -r '.containerImage' <<< "$COMPONENT")
+          if ! [[ "$IMAGE_REF" =~ ^[^:]+@sha256:[0-9a-f]+$ ]] ; then
```

Comment:
Would it not make more sense to just perform this check in the loop on line 134 instead of looping through the components twice?

Reply:
I thought it would make sense to have this check at the very beginning, so that we can fail early. I think it also makes the script easier to read, because this block is only doing one thing - checking the image references. But if you think I should move it there, I can do it.

Reply:
This check will fail the pipeline (it has `exit 1`). So I don't think it matters when it happens in the task, as long as it happens. No external systems are touched in this task (meaning, it isn't like we are pushing to quay or anything) we are just running jq in a workspace that will be immediately removed after as the pipelinerun is going to fail. that is why I think it is better to put it in the other for loop instead of having the same loop twice